src/Model/Gallery.php

```diff
@@ -82,7 +82,7 @@ public function getEnabled()
         return $this->enabled;
     }
 
-    public function setUpdatedAt(?\DateTime $updatedAt = null)
+    public function setUpdatedAt(?\DateTimeInterface $updatedAt = null)
```

Comment:
This is a BC break isn't it? 🤔

Reply:
Why?

Reply:
I think it is, but it feel so bad to delay this PR to a major version...
Maybe an upgrade note could be enough

Reply:
@garak In case someone was overriding the function.

```
class A {
    public function foo(\DatetimeInterface $a) {}
}

class B extends A {
    public function foo(\Datetime $a) {}
}
```
is not valid

Reply:
But we can remove the typehint now and trigger an exception like:

/**
 * NEXT_MAJOR: Add the typehint back.
 */
public function setUpdatedAt(/* ?\DateTimeInterface */$updatedAt = null)
{
    trigger deprecation here to warn the user it needs to be an DateTimeInterface?
    // ...
}

Reply:
@OskarStark but how can you check for argument? You can't check for interfaces

Reply:
What do you mean @garak? IMO we should not do this (as it is right now, introducing a BC break) in the stable branch, these models are meant to be extended by the user.

Reply:
```
class A {
    public function foo($a) {}
}

class B extends A {
    public function foo(\Datetime $a) {}
}
```
Is not more valid IMHO @OskarStark 

> IMO we should not do this (as it is right now, introducing a BC break) in the stable branch, these models are meant to be extended by the user.

Model are meant to be extended but these setters shouldn't be a lot

Reply:
@franmomu I think we can safely change since likely no one is overriding such methods.
I can't see a feasible way to check for argument, in the case we remove the type hint.

Reply:
This change will break my project(s)!

> Model are meant to be extended but these setters shouldn't be a lot

For a greenfield project, you're right, This model classes were designed for extensions, (e.g. the entity class will do this).

Reply:
It is, but since 7.4: https://3v4l.org/iCa85

Reply:
So should we do this PR on master ? :/

Reply:
Yes, that's the only way :(